puppet-checks/src/main/java/com/iadams/sonarqube/puppet/checks/FileModeCheck.java

```diff
@@ -60,22 +63,36 @@ public void init() {
 
   @Override
   public void visitNode(AstNode node) {
-    if(node.getTokenValue().equals("file")) {
+    if ("file".equals(node.getTokenValue())) {
       for (AstNode body : node.getDescendants(PuppetGrammar.RESOURCE_INST)) {
```

Comment:
After having a look at the grammar, I now understand why you extensively use `getDescendants` instead of `getChildren`. It's because the generated AST is too verbose. Quite a few `skip()` should  be added to the grammar. See for example my changes on the grammar in the following commit: https://github.com/iwarapter/sonar-puppet/commit/5ee0c4712ac1e9adbe5074281f0e2d699195939c